## Data Anggota Penyelenggara APJII 

In [1]:
import requests
from bs4 import BeautifulSoup as soup
import csv
from tqdm import tqdm

In [10]:
header = ['no_reg', 'nama_perusahaan', 'nama_merek_usaha', 'tipe_keanggotaan', 'jenis_perizinan', 'domain', 'alamat_kantor',
          'kelurahan', 'kecamatan', 'kabkota', 'provinsi', 'kodepos', 'phone', 'fax']

with open('data/apjii/penyelenggara_apjii.csv', 'w', encoding='UTF8', newline='') as csvfile:
  csv.writer(csvfile).writerow(header)
  

In [2]:
s = requests.Session()

In [3]:
r = s.get('https://apjii.or.id/anggota/penyelengara')
res = soup(r.text, 'html.parser')

In [12]:
pages = int(res.find('div', {'class':'info'}).text.split(' of ')[1].strip())
pages

3798

In [11]:
for page in tqdm(range(1,pages+1)):
  r = s.get(f'https://apjii.or.id/anggota/penyelengara?page={page}')
  res = soup(r.text, 'html.parser')
  
  for content in res.find_all('div', {'class': 'modal-content'}):
    nama_perusahaan = content.find('h5').text
    no_reg = content.find_all('li')[1].find_all('span')[1].text.replace(': ', '').strip()
    nama_merek_usaha = content.find_all('li')[2].find_all('span')[1].text.replace(': ', '').strip()
    tipe_keanggotaan = content.find_all('li')[3].find_all('span')[1].text.replace(': ', '').strip()
    jenis_perizinan = content.find_all('li')[4].find_all('span')[1].text.replace(': ', '').strip()
    domain = content.find_all('li')[5].find_all('span')[1].text.replace(': ', '').strip()
    alamat_kantor = content.find_all('li')[6].find_all('span')[1].text.replace(': ', '').strip()
    kelurahan = content.find_all('li')[7].find_all('span')[1].text.replace(': ', '').strip()
    kecamatan = content.find_all('li')[8].find_all('span')[1].text.replace(': ', '').strip()
    kabkota = content.find_all('li')[9].find_all('span')[1].text.replace(': ', '').strip()
    provinsi = content.find_all('li')[10].find_all('span')[1].text.replace(': ', '').strip()
    kodepos = content.find_all('li')[11].find_all('span')[1].text.replace(': ', '').strip()
    phone = content.find_all('li')[12].find_all('span')[1].text.replace(': ', '').strip()
    fax = content.find_all('li')[13].find_all('span')[1].text.replace(':', '').strip()
    
    data = [no_reg, nama_perusahaan, nama_merek_usaha, tipe_keanggotaan, jenis_perizinan, domain, alamat_kantor,
            kelurahan, kecamatan, kabkota, provinsi, kodepos, phone, fax]
    
    with open('data/apjii/penyelenggara_apjii.csv', mode='a+', newline='', encoding='UTF8') as csvfile:
      csv.writer(csvfile).writerow(data)
  
  

100%|██████████| 3798/3798 [14:58<00:00,  4.23it/s] 


Sebanyak 3798 laman berhasil diakuisi data menggunakan `tqdm` selama 14m 59.1s

## IDNIC - APJII

In [14]:
header = ['no_reg', 'nama_perusahaan', 'nama_merek_usaha', 'tipe_keanggotaan', 'jenis_perizinan', 'domain', 'alamat_kantor',
          'kelurahan', 'kecamatan', 'kabkota', 'provinsi', 'kodepos', 'phone', 'fax']

with open('data/apjii/idnic-apjii.csv', 'w', encoding='UTF8', newline='') as csvfile:
  csv.writer(csvfile).writerow(header)

In [17]:
r = s.get('https://apjii.or.id/anggota/idnic')
res = soup(r.text, 'html.parser')
pages = int(res.find('div', {'class':'info'}).text.split(' of ')[1].strip())
pages

1853

In [15]:
for page in tqdm(range(1,pages+1)):
  r = s.get(f'https://apjii.or.id/anggota/idnic?page={page}')
  res = soup(r.text, 'html.parser')
  
  for content in res.find_all('div', {'class': 'modal-content'}):
    nama_perusahaan = content.find('h5').text
    no_reg = content.find_all('li')[1].find_all('span')[1].text.replace(': ', '').strip()
    nama_merek_usaha = content.find_all('li')[2].find_all('span')[1].text.replace(': ', '').strip()
    tipe_keanggotaan = content.find_all('li')[3].find_all('span')[1].text.replace(': ', '').strip()
    jenis_perizinan = content.find_all('li')[4].find_all('span')[1].text.replace(': ', '').strip()
    domain = content.find_all('li')[5].find_all('span')[1].text.replace(': ', '').strip()
    alamat_kantor = content.find_all('li')[6].find_all('span')[1].text.replace(': ', '').strip()
    kelurahan = content.find_all('li')[7].find_all('span')[1].text.replace(': ', '').strip()
    kecamatan = content.find_all('li')[8].find_all('span')[1].text.replace(': ', '').strip()
    kabkota = content.find_all('li')[9].find_all('span')[1].text.replace(': ', '').strip()
    provinsi = content.find_all('li')[10].find_all('span')[1].text.replace(': ', '').strip()
    kodepos = content.find_all('li')[11].find_all('span')[1].text.replace(': ', '').strip()
    phone = content.find_all('li')[12].find_all('span')[1].text.replace(': ', '').strip()
    fax = content.find_all('li')[13].find_all('span')[1].text.replace(':', '').strip()
    
    data = [no_reg, nama_perusahaan, nama_merek_usaha, tipe_keanggotaan, jenis_perizinan, domain, alamat_kantor,
            kelurahan, kecamatan, kabkota, provinsi, kodepos, phone, fax]
    
    with open('data/apjii/idnic-apjii.csv', mode='a+', newline='', encoding='UTF8') as csvfile:
      csv.writer(csvfile).writerow(data)
  
  

100%|██████████| 1853/1853 [05:59<00:00,  5.15it/s]


Sebanyak 1853 laman berhasil diakuisi data menggunakan `tqdm` selama 5m 59.6s